### DISCUSSION TITLE
Blockprogrammierung Python in Blöcke umsetzen

### DISCUSSION THREAD

**Discussion by msw-home** [pybricks/support#1422] [Q&A]
Moin zusammen.
Kann mir jemand helfen, dieses kleine Programm in Blöcken zu programmieren?
Ist ein Fahrzeug 1 Antrieb, 1 Lenkung, 1 FB, manuelle Umschaltung für die Geschwindigkeit.
Würde mir vielleicht helfen einiges zu verstehen.

LG Manfred

from pybricks.pupdevices import *
from pybricks.parameters import *
from pybricks.tools import wait

steering = Motor(Port.B)
driving = Motor(Port.A)
remoteControl = Remote()

#steering calibration sequence
steering.run_until_stalled(500)
steering.reset_angle(0)
steering.run_until_stalled(-500)
maxAngle = steering.angle()
steering.run_target(500, (maxAngle/2)+10)
steering.reset_angle(0)

#main loop
while True:
    #update pressed buttons information
    pressed = remoteControl.buttons.pressed()

    #commands to drive forwards or backwards
    if Button.LEFT_PLUS in pressed:
        driving.dc(100)
    elif Button.LEFT_MINUS in pressed:
        driving.dc(-100)
    else: 
        driving.brake()

    #commands to steer
    if Button.RIGHT_PLUS in pressed:
        steering.run_target(1200, (maxAngle/2)-5, Stop.HOLD,False)
    elif Button.RIGHT_MINUS in pressed:
        steering.run_target(1200, -((maxAngle/2)-5), Stop.HOLD,False)
    else:
        steering.run_target(1200, 0, Stop.HOLD, False)

    wait(100)

**Comment by laurensvalk** [comment]
We just released new blocks to make this a lot simpler. Check out this video which includes just the program you are looking for.

You can try out the new functionality:
- In Pybricks, go to settings.
- Click "try new beta features"
- This opens up a test version of Pybricks, which already contains the new features.

This should come to the normal version soon as well.

[![image](https://github.com/pybricks/support/assets/12326241/4c95d5c2-8c4c-4c11-9cb3-0a6e1b2995e8)](https://www.youtube.com/watch?v=msru8kz5tR4)

**Reply by msw-home** [reply]
Super, danke, das hilft mir weiter!
